### DISCUSSION TITLE
SimilarityMaps.GetSimilarityMapFromWeights

### DISCUSSION THREAD

**Discussion by PARODBE** [rdkit/rdkit#4314] [Q&A]
Hi,

I have a question. When you use this function and you obtain a similaritymap with greener for the part of molecule with a higher contribution for similarity:
![image](https://user-images.githubusercontent.com/62376157/124884736-da170500-dfd2-11eb-84de-f1afa97cc7bb.png)

How you can extract the smiles of the different parts colored?

Thanks!

All Best!

Pablo

**Comment by greglandrum** [comment]
Can you please be more specific about exactly what you want to do?

**Comment by PARODBE** [comment]
yes, thank you for answering, I would like create a dataframe with the fragments of each part of the molecule. For example, imagine that you have built a similaritymaps with a Machine Learning model for toxicity prediction of the substructures, like this: 

![image](https://user-images.githubusercontent.com/62376157/125456747-f8181961-d8b5-4a8d-83dd-5e27cc48a19c.png)

And I would want build a dataframe with substructures colored in blue and substructures colored in red, and for this dataframe I would like have smiles of the substructures.

Thanks!

Pablo

**Reply by greglandrum** [reply]
If you can identify the set of atoms which you are interested in (perhaps by setting a threshold value for weights which you consider to be high enough), then you can get the SMILES for those atoms of the molecule using `MolFragmentToSmiles()`:
```
In [17]: m = Chem.MolFromSmiles('CCc1ccc(CO)cc1')

In [18]: Chem.MolFragmentToSmiles(m,atomsToUse=[1,2,3,4])
Out[18]: 'cccC'
```
Depending upon what you're interested in, it may make more sense to ensure that information about H counts is included:
```
In [19]: Chem.MolFragmentToSmiles(m,atomsToUse=[1,2,3,4],allHsExplicit=True)
Out[19]: '[cH][cH][c][CH2]'
```
If your collection of "interesting" atoms includes multiple fragments, then this SMILES will have a series of elements separated by `.` charactes. You can easily get the individual fragments by splitting:
```
In [22]: Chem.MolFragmentToSmiles(m,atomsToUse=[1,2,3,5,6,7],allHsExplicit=True).split('.')
Out[22]: ['[c][CH2][OH]', '[cH][c][CH2]']
```

**Reply by PARODBE** [reply]
Thanks ! very useful!
